src/Braket.jl

```diff
@@ -134,6 +139,139 @@ end
+    subbed_expr = Symbolics.substitute(fpe.expression, param_values_num)
+    if isempty(Symbolics.get_variables(subbed_expr))
+	   subbed_expr = Symbolics.value(subbed_expr)
+        return subbed_expr
+    else
+	subbed_expr = Symbolics.value(subbed_expr)
+	return FreeParameterExpression(subbed_expr)
+    end 
+end
+
+function Base.:+(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1.expression + fpe2.expression)
+end
+
+function Base.:+(fpe1::FreeParameterExpression, fpe2::Union{Number, Symbolics.Num})
+    return FreeParameterExpression(fpe1.expression + fpe2)
+end
+
+function Base.:+(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1 + fpe2.expression)
+end
+
+function Base.:*(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1.expression * fpe2.expression)
+end
+
+function Base.:*(fpe1::FreeParameterExpression, fpe2::Union{Number, Symbolics.Num})
+    return FreeParameterExpression(fpe1.expression * fpe2)
+end
+
+function Base.:*(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1 * fpe2.expression)
+end
+
+function Base.:-(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1.expression - fpe2.expression)
+end
+
+function Base.:-(fpe1::FreeParameterExpression, fpe2::Union{Number, Symbolics.Num})
+    return FreeParameterExpression(fpe1.expression - fpe2)
+end
+
+function Base.:-(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1 - fpe2.expression)
+end
+
+function Base.:/(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1.expression / fpe2.expression)
+end
+
+function Base.:/(fpe1::FreeParameterExpression, fpe2::Union{Number, Symbolics.Num})
+    return FreeParameterExpression(fpe1.expression / fpe2)
+end
+
+function Base.:/(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1 / fpe2.expression)
+end
+
+function Base.:^(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1.expression ^ fpe2.expression)
+end
+
+function Base.:^(fpe1::FreeParameterExpression, fpe2::Union{Number, Symbolics.Num})
+    return FreeParameterExpression(fpe1.expression ^ fpe2)
+end
+
+function Base.:^(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression)
+    return FreeParameterExpression(fpe1 ^ fpe2.expression)
+end
+
+function Base.:(==)(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+    return isequal(fpe1.expression, fpe2.expression)
+end
+
+function Base.:!=(fpe1::FreeParameterExpression, fpe2::FreeParameterExpression)
+     return !(isequal(fpe1.expression, fpe2.expression))
+end
+
```

Comment:
Could slim all these down by making them one liners like:

```
Base.:^(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression) = FreeParameterExpression(fpe1 ^ fpe2.expression)
```

Reply:
This was giving error

```
Base.:^(fpe1::Union{Number, Symbolics.Num}, fpe2::FreeParameterExpression) = FreeParameterExpression(fpe1 ^ fpe2.expression)

julia> result = fpe1 ^ fpe2
ERROR: MethodError: no method matching ^(::FreeParameterExpression, ::FreeParameterExpression)

Closest candidates are:
  ^(::Number, ::FreeParameterExpression)
   @ Braket ~/Desktop/New/braket/2/Braket.jl/src/Braket.jl:211
  ^(::SymbolicUtils.Symbolic{<:Number}, ::Any)
   @ SymbolicUtils ~/.julia/packages/SymbolicUtils/qyMYa/src/types.jl:1179

```
We can slim it down like this:
```

Base.:+(fpe1::FreeParameterExpression, fpe2::Union{FreeParameterExpression, Number, Symbolics.Num}) = 
    FreeParameterExpression((fpe1 isa FreeParameterExpression ? fpe1.expression : fpe1) + 
                            (fpe2 isa FreeParameterExpression ? fpe2.expression : fpe2))

```